### PR TITLE
[3505] Create instance, holdings, and item records for a print resource (folijet) (TaaS)

### PR DESCRIPTION
**Pull request by dnovikau**
[C3505](https://foliotest.testrail.io/index.php?/cases/view/3505)
***
[Allure report - test branch](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5036/allure)
[Allure report - master branch](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5039/allure)
***

### CODE REVIEW

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    it(
+      'C3505 Create instance, holdings, and item records for a print resource which has not been acquired through Orders (folijet) (TaaS)',
+      { tags: [TestTypes.extendedPath, DevTeams.folijet] },
+      () => {
+        // Click on "New" in the "Actions" menu
+        const InventoryNewInstance = InventoryInstances.addNewInventory();
+
+        // Populate the required fields, Click 'Save and close'
+        InventoryNewInstance.fillInstanceFields({
```
Comment (manowarqqq) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:38]: Not all fields from TC are filled

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          statusTerm: 'Cataloged (folio: cat)',
+          mode: 'integrating resource',
+        });
+        InventoryNewInstance.clickSaveAndCloseButton();
```
Comment (manowarqqq) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:42]: Need to check Source: FOLIO

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        // Select "Add holdings"
+        const HoldingsRecordEdit = InventoryInstance.openEditHoldingsForm();
```
Comment (manowarqqq) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:45]: There is already InventoryInstance.pressAddHoldingsButton() method

File: cypress/support/fragments/inventory/inventoryInstance.js [hunk 4]
```diff
@@ -617,6 +620,10 @@ export default {
   clickAddItemByHoldingName(holdingName) {
     const holdingSection = section.find(Accordion(including(holdingName)));
     cy.do(holdingSection.find(addItemButton).click());
+
+    ItemRecordEdit.waitLoading('');
```
Comment (manowarqqq) [inline cypress/support/fragments/inventory/inventoryInstance.js:624]: This should be Instance resourse_title name not empty string

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        // Select "Add item"
+        const ItemRecordEdit = InventoryInstance.clickAddItemByHoldingName(testData.callNumber);
+
+        // Populate the following fields: "Material type", "Permanent loan type"
+        // Click on "Save & Close" button
+        ItemRecordEdit.fillItemRecordFields({
+          materialType: 'book',
+          loanType: 'Can circulate',
+        });
+        ItemRecordEdit.saveAndClose({ itemSaved: true });
```
Comment (manowarqqq) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:64]: Need to check item count from Step 6 expected result
Accordion(...).find(Badge()).has({ value: value.toString() })
Updated: Already exists method verifyNumberOfItemsInHoldingByName

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        testData.user = userProperties;
+
+        cy.login(testData.user.username, testData.user.password, {
+          path: TopMenu.inventoryPath,
+          waiter: InventoryInstances.waitContentLoading,
+        });
+      });
+    });
+
+    after('Delete test data', () => {
```
Comment (manowarqqq) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:26]: Need to remove instance, holding,  item after test

File: cypress/support/fragments/inventory/inventoryInstance.js (hunk 4 above)
Comment (dnovikau) [reply cypress/support/fragments/inventory/inventoryInstance.js:624]: Done

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js (hunk 3 above)
Comment (dnovikau) [reply cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:45]: Fixed

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js (hunk 2 above)
Comment (dnovikau) [reply cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:42]: Done

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js (hunk 6 above)
Comment (dnovikau) [reply cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:26]: Done

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js (hunk 1 above)
Comment (dnovikau) [reply cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:44]: Done

**Review by manowarqqq — APPROVED** [review]

File: cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+import TopMenu from '../../../support/fragments/topMenu';
+import InventoryInstance from '../../../support/fragments/inventory/inventoryInstance';
+import InventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
+import Users from '../../../support/fragments/users/users';
+
+describe('Inventory', () => {
+  describe('Cataloging', () => {
+    const testData = {
+      callNumber: '331.2',
+      barcode: getRandomPostfix(),
```
Comment (TetianaParanich) [inline cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js:15]: It's better use uuid() for as random barcode

Comment (dnovikau) [reply cypress/e2e/inventory/cataloging/create-instance-holdings-and-item.cy.js]: Done

**Pull request merged**